packages/fxa-auth-server/lib/payments/paypal/helper.ts

```diff
@@ -455,15 +455,15 @@ export class PayPalHelper {
     // attempt in combination, so that retries can be made if
     // the prior attempt failed and a retry is desired.
     const idempotencyKey = this.generateIdempotencyKey(
-      invoice.id,
+      invoice.id!,
```

Comment:
Any details on what the breaking change is in the API to cause this? I found this but not sure, https://stripe.com/docs/upgrades#2019-12-03

Reply:
I don't think these were caused by the latest version of Stripe - the problems already existed while using v9.6.0, as `id` is an optional property within `Invoice`, which triggered TypeScript.

<img width="918" alt="Screen Shot 2022-09-07 at 2 58 17 PM" src="https://user-images.githubusercontent.com/28129806/188962689-4fad9d8b-a8bb-4d6d-a8aa-66611454d28e.png">

Reply:
Oh in older version (2022-08-01) this says it is required.

<img width="786" alt="Screen Shot 2022-09-08 at 12 47 41 PM" src="https://user-images.githubusercontent.com/1295288/189179854-b7712522-5569-4014-9d61-03d0562d832a.png">

If I understand correctly, this change is saying that `invoice.id` is not null but in the new api version it can be since its optional? It isn't clear to me if this would break any new or old invoices.

Either way I'll defer to someone on subplat for review since they are more familiar with Stripe API 😅.

Reply:
i did a cursory search on google for the id being undefined, which i think only happens when you preview an upcoming invoice, so i think we should be good here? 

For any of the uses here we do need the id, so enforcing that its not undefined should not be an issue.